CONTRIBUTE.md

```diff
@@ -0,0 +1,154 @@
+basics](https://git-scm.com/book/en/v2/Getting-Started-Git-Basics) and
+an [interactive git
+tutorial](https://try.github.io/levels/1/challenges/1).  You must also
+[install
+git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+locally on your computer.
+
+### Forking and cloning PlasmaPy
+
+After creating your GitHub account, go to the [main
+repository](https://github.com/PlasmaPy/PlasmaPy) and **fork a copy of
+PlasmaPy to your account**.
+
+Next you must **clone your fork to your computer**.  Go to the
+directory that will host your PlasmaPy directory, and run one of the
+following commands (after changing *your-username* to your username).
+If you would like to use HTTPS, then run:
+```
+git clone https://github.com/*your-username*/PlasmaPy.git
+```
+If you have previously [set up an SSH
+key](https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/), then you may run:
+```
+git clone git@github.com:*your-username*/PlasmaPy.git
+```
+
+After cloning, we must tell git where the development version of
+PlasmaPy is by running:
+```
+git remote add plasmapy git://github.com/PlasmaPy/PlasmaPy.git
+```
+Another common name for the development version of repositories is `upstream`.
+
+To emphasize which remote is your fork of PlasmaPy, we will rename origin to *your-username* by running:
+```
+git remote rename origin *your-username*
+```
+
+To check on which remotes exist, run `git remote -v`.  You should get
+something like this:
+```
+namurphy	git@github.com:namurphy/PlasmaPy.git (fetch)
+namurphy	git@github.com:namurphy/PlasmaPy.git (push)
+plasmapy	git@github.com:PlasmaPy/PlasmaPy.git (fetch)
+plasmapy	git@github.com:PlasmaPy/PlasmaPy.git (push)
+```
+
+### Branches, commits, and pull requests
+
+Before making any changes, it is prudent to update your local
+repository with the most recent changes from the development
+repository: 
+```
+git fetch plasmapy
+```
+
+Changes to PlasmaPy should be made using branches.  It is usually best
+to avoid making changes on your master branch.  Instead we can create
+a new branch for the specific feature that you would like to work on:
+```
+git branch *your-new-feature*
+```
+Descriptive branch names such as `grad-shafranov` or
+`adding-eigenfunction-poetry` are helpful, while vague names like
+`edits` are considered harmful.  After creating your branch locally,
+let your fork of PlasmaPy know about it by running:
+```
+git push --set-upstream *your-username* *your-new-feature*
+```
+It is also useful to configure git so that only the branch you are
+working on gets pushed to GitHub:
+```
+git config --global push.default simple
+```
+
+Once you have set up your fork and created a branch, you are ready to
+make edits to PlasmaPy.  Switch to your new branch by running:
+```
+git checkout *your-new-feature*
+```
+Go ahead and modify files with your favorite text editor.  Be sure to
+include tests and documentation with any new functionality.  We also
```

Comment:
Yes, definitely.  It would be great if we could have an example set of tests and docs that we could point to in our code.  The example tests should also include how to test to make sure that exceptions come up when needed (as that is something that I need to start doing).